src/pages/Trade/components/OrderBookCard/MyTrades.tsx

```diff
@@ -1,49 +1,110 @@
-import React, { FC, useEffect } from 'react';
+import React, { FC } from 'react';
+import { useQuery } from 'react-query';
 import { connect, ConnectedProps } from 'react-redux';
-
-import {
-	fetchMyTradesRequest,
-	getMyTrades,
-	getIsLoadingMyTrades,
-	getIsLoadedMyTrades,
-} from 'ducks/trades/myTrades';
+import snxData from 'synthetix-data';
+import { SYNTHS_MAP } from 'constants/currency';
 
 import { RootState } from 'ducks/types';
-
-import useInterval from 'shared/hooks/useInterval';
-
+import { getIsWalletConnected, getCurrentWalletAddress } from 'ducks/wallet/walletDetails';
+import { normalizeTrades } from 'ducks/trades/utils';
 import TradeHistory from './TradeHistory';
 
 import { REFRESH_INTERVAL } from './constants';
+import { HistoricalTrades } from 'ducks/trades/types';
+import QUERY_KEYS from 'constants/queryKeys';
+import { CurrencyKey } from 'constants/currency';
+
+const MAX_TRADES = 100;
 
 const mapStateToProps = (state: RootState) => ({
-	trades: getMyTrades(state),
-	isLoading: getIsLoadingMyTrades(state),
-	isLoaded: getIsLoadedMyTrades(state),
+	walletAddress: getCurrentWalletAddress(state),
+	isWalletConnected: getIsWalletConnected(state),
 });
 
-const mapDispatchToProps = {
-	fetchMyTradesRequest,
-};
-
-const connector = connect(mapStateToProps, mapDispatchToProps);
+const connector = connect(mapStateToProps);
 
 type PropsFromRedux = ConnectedProps<typeof connector>;
 
 type MyTradesProps = PropsFromRedux;
 
-const MyTrades: FC<MyTradesProps> = ({ fetchMyTradesRequest, trades, isLoading, isLoaded }) => {
-	useEffect(() => {
-		fetchMyTradesRequest();
-		// eslint-disable-next-line react-hooks/exhaustive-deps
-	}, []);
+type SettledTrade = {
+	exchangeTimestamp: number;
+	dest: CurrencyKey;
+	src: CurrencyKey;
+	amount: number;
+	rebate: number;
+	reclaim: number;
+};
+type SettledTrades = SettledTrade[];
+
+const MyTrades: FC<MyTradesProps> = ({ walletAddress, isWalletConnected }) => {
+	const settledTradesQuery = useQuery<SettledTrades, any>(
+		QUERY_KEYS.HistoricalTrades.AllTrades,
```

Comment:
I had copied the wrong key here earlier, that was the root cause of my fee bug.  Thats what I get for copy/pasting 🤦‍♂️